source/rock/middle/TypeDecl.ooc

```diff
@@ -1360,8 +1383,14 @@ TypeDecl: abstract class extends Declaration {
         if (fDecl) {
             if (call debugCondition()) "    \\o/ Found fDecl for %s, it's %s" format(call name, fDecl toString()) println()
             if (call suggest(fDecl, res, trail)) {
-                if (fDecl hasThis() && !call getExpr()) {
-                    call setExpr(VariableAccess new("this", call token))
+                if (!call getExpr()) {
+                    if (fDecl hasThis()) {
+                        call setExpr(VariableAccess new("this", call token))
```

Comment:
Adding the `this` argument is not done for static functions but by splitting the condition into `missing this` (`!call getExpr()`) and `expect this` (`fDecl hasThis()`), we can give warnings for static functions too.